### DISCUSSION TITLE
Control - C to Abort osxphotos export run - Not Working ?

### DISCUSSION THREAD

**Discussion by rajscode** [RhetTbull/osxphotos#1431] [Q&A] [answered]
Have been confronting an issue where a CONTROL - C (1 or many times even) to abort an osxphotos export fails to terminate the run? This use to work without issue on 0.64.x and prior releases. Definitely hasn't worked since I upgraded to 0.67.x releases for me. Wondering if it could be releated to python 3.11 / 3.12 ?   I eventually have to kill the process id for osxphotos.  

Environment I am running:
osxphotos, version 0.67.4
Python 3.12.2 (main, Feb  6 2024, 20:19:44) [Clang 15.0.0 (clang-1500.1.0.2.5)]
macOS 14.3.1, arm64

Stored on an external SSD (apfs)
osxphotos export db is v9.1 and a size of 2.74 Gigabytes
photos library is v 6000,9. and a size of 1.66 Terabytes
tmpdir is on same volume

```shell
osxphotos export /Volumes/WDSSD2TB/photo_library_export/photo_library_backup/ --ramdb --download-missing --use-photokit --load-config osxphotos_backup.toml
``` 
<details><summary>contents of osxphotos_backup.toml</summary>
<p>

[export]
update = true
cleanup = true
report = "/Volumes/WDSSD2TB/photo_library_export/osxphotos_report_backup.sqlite"
append = true
db = "/Volumes/WDSSD2TB/Macbook_Pro_14_PhotosLibrary_Full_Originals_Backup/Photos Library.photoslibrary"
tmpdir = "/Volumes/WDSSD2TB/photo_library_export/osxphotos_tmp"

directory = "{created.year}/{created.year}-{created.mm}"
theme = "dark"

sidecar = [ "xmp",]
sidecar_drop_ext = true

person_keyword = true
favorite_rating = true
keyword_template = [
    "Album: {folder_album}",
    "{label}",
    "{favorite?is_favorite}",
    "is_{media_type}",
]

</p>
</details> 

**Result of export command abort attempt:**
```shell
2024-03-04 15:56:43,738 - root - WARNING - photosdb.py - 157 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4, 11.5, 11.6, 11.7, 12.0, 12.1, 12.2, 12.3, 12.4, 12.5, 12.6, 12.7, 13.0, 13.1, 13.2, 13.3, 13.4, 13.5, 13.6, 14.0, 14.1]: you have Darwin, OS version: 14.3
Processing database /Volumes/WDSSD2TB/Macbook_Pro_14_PhotosLibrary_Full_Originals_Backup/Photos Library.photoslibrary/database/photos.db
Processing database /Volumes/WDSSD2TB/Macbook_Pro_14_PhotosLibrary_Full_Originals_Backup/Photos Library.photoslibrary/database/Photos.sqlite
...
...
Processing photo details.
Processing import sessions.
Processing additional photo details.
**^CAborted!**
Writing export database to /Volumes/WDSSD2TB/photo_library_export/photo_library_backup/.osxphotos_export.db
**Aborted!**
Writing export database to /Volumes/WDSSD2TB/photo_library_export/photo_library_backup/.osxphotos_export.db
**zsh: killed**     /Users/rajscode/.local/bin/osxphotos export  --ramdb  --use-photokit
```

**Comment by RhetTbull** [comment]
I've been able to reproduce this. Definitely a bug so I'll create an issue from this.  The problem is here in `osxphotos/cli/export.py::export_cli()`:

```python
        if ramdb:
            # if using ramdb, ensure database is written in case of crash
            export_db_callback = register_crash_callback(
                export_db.write_to_disk, f"Writing export database to {export_db_path}"
            )

            # install SIGINT handler to write database on Ctrl+C
            def sigint_handler(signal, frame):
                print("Aborted!", file=sys.stderr)
                print(f"Writing export database to {export_db_path}")
                export_db.write_to_disk()
                return 1

            signal.signal(signal.SIGINT, sigint_handler)
```

The SIGINT handler is installed to catch Ctrl+C to ensure the export database is written back to disk in the case of Ctrl+C. This was a possible source of data loss if user interrupted an export when using `--ramdb`, the export database could be left in a state where it didn't match the export folder. I tested this when first done but something's changed and it's definitely not aborting the code now.

**Reply by RhetTbull** [reply]
I pushed a hot fix in 0.67.6 -- update and this should be resolved.

**Reply by RhetTbull** [reply, marked as answer]
Issue was `return 1` should be `sys.exit(1)`, fixed in v0.67.6
